test/source.test.ts

```diff
@@ -355,6 +355,61 @@ describe("source accessor in fields", () => {
     disposeB();
   });
 
+  test("form with autoLoad disposes on delete", async () => {
+    const n = types.model("N", {
+      foo: types.array(M),
+      containerA: ContainerA,
+      containerB: ContainerB
+    });
+
+    const r = n.create({
+      foo: [
+        {
+          a: undefined,
+          b: undefined
+        }
+      ],
+      containerA: { entryMap: {} },
+      containerB: { entryMap: {} }
+    });
+
+    const containerA = r.containerA;
+    const containerB = r.containerB;
+
+    const sourceA = new Source({ entryMap: containerA.entryMap, load: loadA });
+    const sourceB = new Source({ entryMap: containerB.entryMap, load: loadB });
+
+    const form = new Form(n, {
+      foo: new RepeatingForm({
+        a: new Field(converters.maybe(converters.model(ItemA)), {
+          references: {
+            source: sourceA
+          }
+        }),
+        b: new Field(converters.maybe(converters.model(ItemB)), {
+          references: {
+            source: sourceB,
+            // this source is dependent on state. This information
+            // should be sent whenever a load is issued
+            dependentQuery: accessor => {
+              return { a: accessor.node.a, c: accessor.node.c };
+            }
+          }
+        })
+      })
+    });
+    const state = form.state(r);
+    const repeating = state.repeatingForm("foo");
+    const firstLine = repeating.index(0);
+
+    const fieldA = firstLine.field("a");
+    const fieldB = firstLine.field("b");
+    fieldA.references.autoLoadReaction();
+    fieldB.references.autoLoadReaction();
+
+    repeating.remove(r.foo[0]);
```

Comment:
I see no assertions in this test. Does this test that the remove works without errors?

Reply:
Good catch! I've added a counter to check how often the dependentQuery gets called. We don't except the query to be called on remove.